src/components/tooltip/tooltip.component.js

```diff
@@ -1,70 +1,97 @@
-import React from "react";
+import React, { useRef } from "react";
 import PropTypes from "prop-types";
-import { StyledTooltipInner, StyledTooltipWrapper } from "./tooltip.style";
-import StyledTooltipPointer from "./tooltip-pointer.style";
-import OptionsHelper from "../../utils/helpers/options-helper/options-helper";
+import Tippy from "@tippyjs/react/headless";
+import StyledTooltip from "./tooltip.style";
+import StyledPointer from "./tooltip-pointer.style";
 import tagComponent from "../../utils/helpers/tags/tags";
 
-class Tooltip extends React.Component {
-  static propTypes = {
-    /**
-     * Sets alignment of pointer on tooltip
-     */
-    align: PropTypes.oneOf(OptionsHelper.alignAroundEdges),
-    /** Children elements */
-    children: PropTypes.node,
-    /** The id attribute to use for the tooltip */
-    id: PropTypes.string,
-    /** Whether to to show the Tooltip */
-    isVisible: PropTypes.bool,
-    /**
-     * Sets position of the tooltip
-     */
-    position: PropTypes.oneOf(OptionsHelper.positions),
-    /** Sets a onMouseEnter function */
-    onMouseEnter: PropTypes.func,
-    /** Sets a onMouseLeave function */
-    onMouseLeave: PropTypes.func,
-    /** Defines the message type */
-    type: PropTypes.string,
-  };
+const Tooltip = React.forwardRef(
+  (
+    {
+      children,
+      isVisible,
+      position = "top",
+      message,
+      type,
+      size = "M",
+      isPartOfInput,
+      inputSize,
+      id,
+      ...props
+    },
+    ref
+  ) => {
+    const tooltipRef = useRef(ref || null);
 
-  static defaultProps = {
-    align: "center",
-    position: "top",
-    isVisible: false,
-  };
+    const tooltip = (attrs, content) => {
+      const currentPosition = attrs["data-placement"] || position;
+      const positionArrowAbove = ["bottom", "right"].includes(currentPosition);
 
-  get tooltipHTML() {
-    const { children, onMouseEnter, onMouseLeave, ...commonProps } = this.props;
+      const pointer = (
+        <StyledPointer
+          key="pointer"
+          type={type}
+          {...attrs}
+          position={currentPosition}
+          data-popper-arrow=""
```

Comment:
Is this meant to be an empty string?

Reply:
Yeah that's how it is in the docs https://github.com/atomiks/tippyjs-react#headless-tippy